diesel/src/sqlite/connection/functions.rs

```diff
@@ -53,6 +53,32 @@ where
     Ok(())
 }
 
+pub fn register_aggregate<ArgsSqlType, RetSqlType, Args, Ret, A>(
+    conn: &RawConnection,
+    fn_name: &str,
+) -> QueryResult<()>
+where
+    A: Aggregator<Args, Output=Ret> + Send,
+    Args: Queryable<ArgsSqlType, Sqlite>,
+    Ret: ToSql<RetSqlType, Sqlite>,
+    Sqlite: HasSqlType<RetSqlType>,
+{
+    let fields_needed = Args::Row::FIELDS_NEEDED;
+    if fields_needed > 127 {
+        return Err(Error::DatabaseError(
+            DatabaseErrorKind::UnableToSendCommand,
+            Box::new("SQLite functions cannot take more than 127 parameters".to_string()),
+        ));
+    }
+
+    // TODO @thekuom: how do we convert this to ffi::sqlite3_value?
```

Comment:
In the `register` function, the function passed into the connection takes `ffi::sqlite3_value` as args. I am trying to do the same here. I haven't put too much time into this part yet, but I would appreciate any advice on approaching this function.